src/util/market_prices.ts

```diff
@@ -32,6 +38,35 @@ export const getZeroXPriceInUSD = async (): Promise<BigNumber> => {
     return zeroXPriceInUSD;
 };
 
+export const getZeroXPriceInWeth = async (): Promise<BigNumber> => {
+    /* Gets all the available sell orders of 0x and takes the lowest price, is there are no 0x sells orders available, the price is calculated using coin market */
+    const web3Wrapper = await getWeb3Wrapper();
+    let zeroXPrice = new BigNumber('0');
+    if (web3Wrapper) {
+        const networkId = await web3Wrapper.getNetworkIdAsync();
+        const knownTokens = getKnownTokens(networkId);
+        const zeroXToken = knownTokens.getTokenBySymbol('ZRX');
+        let sellOrders = await getAllOrdersAsUIOrders(zeroXToken);
+        if (sellOrders && sellOrders.length > 0) {
+            sellOrders = sellOrders
+                .filter(order => order.side === OrderSide.Sell)
+                .sort((o1, o2) => o1.price.comparedTo(o2.price));
+            /* Check if there is any order that could fill the needs of 0x */
+            const orderWithEnoughZeroX = sellOrders.find(sellOrder => {
```

Comment:
Q: What is the reason for this? In the (unlikely) case where the cheapest sell order has a size that is less than the fee amount, why wouldn't that price be a good enough estimation?

Reply:
I'm not very sure if this is a good of wrong estimation. I'd discussed it with @unjapones and we decided to go for the "check the market price" approach, but we could just remove it and use the orderbook value. What do you guys think?